segmentation fault

I install tensorflow use:sudo pip install --upgrade https://storage.googleapis.com/tensorflow/linux/gpu/tensorflow-0.8.0rc0-cp27-none-linux_x86_64.whl

linux version is :Linux version 3.10.0-229.el7.x86_64 (builder@kbuilder.dev.centos.org) (gcc version 4.8.2 20140120 (Red Hat 4.8.2-16) (GCC) ) #1 SMP Fri Mar 6 11:36:42 UTC 2015


when I learn python, and import tensorflow,segmentation fault

>>> import tensorflow
I tensorflow/stream_executor/dso_loader.cc:105] successfully opened CUDA library libcublas.so locally
I tensorflow/stream_executor/dso_loader.cc:105] successfully opened CUDA library libcudnn.so locally
I tensorflow/stream_executor/dso_loader.cc:105] successfully opened CUDA library libcufft.so locally
I tensorflow/stream_executor/dso_loader.cc:105] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:105] successfully opened CUDA library libcurand.so locally
Segmentation fault (core dumped)

I noticed something potentially relevant here:

```
INFO: From Compiling tensorflow/python/lib/core/py_func.cc:
In file included from third_party/py/numpy/numpy_include/numpy/ndarraytypes.h:17
77:0,
                 from third_party/py/numpy/numpy_include/numpy/ndarrayobject.h:1
8,
                 from third_party/py/numpy/numpy_include/numpy/arrayobject.h:4,
                 from tensorflow/python/lib/core/py_func.cc:19:
third_party/py/numpy/numpy_include/numpy/npy_1_7_deprecated_api.h:15:2: warning:
 #warning "Using deprecated NumPy API, disable it by " "#defining NPY_NO_DEPRECA
TED_API NPY_1_7_API_VERSION" [-Wcpp]
 #warning "Using deprecated NumPy API, disable it by " \
  ^
```

There's a warning about the numpy include using deprecated NumPy API.

I tried to use `#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION` as it is used in other TensorFlow headers, but then the project fails to compile:

```
tensorflow/python/lib/core/py_func.cc: In function 'tensorflow::Status tensorflow::{anonymous}::ConvertNdarrayToTensor(PyObject*, tensorflow::Tensor*)':
tensorflow/python/lib/core/py_func.cc:235:50: error: 'PyArrayObject' has no member named 'data'
       memcpy(const_cast<char*>(p.data()), input->data, p.size());
                                                  ^
tensorflow/python/lib/core/py_func.cc: In function 'tensorflow::Status tensorflow::ConvertTensorToNdarray(const tensorflow::Tensor&, PyObject**)':
tensorflow/python/lib/core/py_func.cc:330:61: error: 'PyArrayObject' has no member named 'data'
     PyObject** out = reinterpret_cast<PyObject**>(np_array->data);
                                                             ^
tensorflow/python/lib/core/py_func.cc:345:22: error: 'PyArrayObject' has no member named 'data'
     memcpy(np_array->data, p.data(), p.size());
                      ^
```

Is TensorFlow trying to use two different versions of the NumPy API?

It does look like an old numpy is creeping in from somewhere.
On Tue, Apr 26, 2016 at 06:48 Trevor Simonton <notifications@github.com>
wrote:

> I noticed something potentially relevant here:
>
> INFO: From Compiling tensorflow/python/lib/core/py_func.cc:
> In file included from third_party/py/numpy/numpy_include/numpy/ndarraytypes.h:17
> 77:0,
>                  from third_party/py/numpy/numpy_include/numpy/ndarrayobject.h:1
> 8,
>                  from third_party/py/numpy/numpy_include/numpy/arrayobject.h:4,
>                  from tensorflow/python/lib/core/py_func.cc:19:
> third_party/py/numpy/numpy_include/numpy/npy_1_7_deprecated_api.h:15:2: warning:
>  #warning "Using deprecated NumPy API, disable it by " "#defining NPY_NO_DEPRECA
> TED_API NPY_1_7_API_VERSION" [-Wcpp]
>  #warning "Using deprecated NumPy API, disable it by " \
>   ^
>
> There's a warning about the numpy include using deprecated NumPy API.
>
> I tried to use #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION as it is
> used in other TensorFlow headers, but then the project fails to compile:
>
> tensorflow/python/lib/core/py_func.cc: In function 'tensorflow::Status tensorflow::{anonymous}::ConvertNdarrayToTensor(PyObject*, tensorflow::Tensor*)':
> tensorflow/python/lib/core/py_func.cc:235:50: error: 'PyArrayObject' has no member named 'data'
>        memcpy(const_cast<char*>(p.data()), input->data, p.size());
>                                                   ^
> tensorflow/python/lib/core/py_func.cc: In function 'tensorflow::Status tensorflow::ConvertTensorToNdarray(const tensorflow::Tensor&, PyObject**)':
> tensorflow/python/lib/core/py_func.cc:330:61: error: 'PyArrayObject' has no member named 'data'
>      PyObject** out = reinterpret_cast<PyObject**>(np_array->data);
>                                                              ^
> tensorflow/python/lib/core/py_func.cc:345:22: error: 'PyArrayObject' has no member named 'data'
>      memcpy(np_array->data, p.data(), p.size());
>                       ^
>
> Is TensorFlow trying to use two different versions of the NumPy API?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/issues/2034#issuecomment-214750702>
>

--- a/tensorflow/python/__init__.py
+++ b/tensorflow/python/__init__.py
@@ -40,6 +40,10 @@ import traceback
 # the mode to RTLD_GLOBAL to make the symbols visible, so libraries such
 # as the ones implementing custom ops can have access to tensorflow
 # framework's symbols.
+# one catch is that numpy *must* be imported before the call to
+# setdlopenflags(), or there is a risk that later c modules will segfault
+# when importing numpy (gh-2034).
+import numpy as np
 _default_dlopen_flags = sys.getdlopenflags()
 sys.setdlopenflags(_default_dlopen_flags | ctypes.RTLD_GLOBAL)
 from tensorflow.python import pywrap_tensorflow

